### DISCUSSION TITLE
Switch to using octokit

### DISCUSSION THREAD

**Discussion by mAmineChniti** [AminoffZ/github-repo-size#40] [Ideas]
We can use Octokit.js to interact with repos and fetch all kinds of information instead of wring our own fetch requests
[Quickstart for GitHub REST API](https://docs.github.com/en/rest/quickstart?apiVersion=2022-11-28&tool=javascript)
[Octokit Repo](https://github.com/octokit/octokit.js/#readme)

**Comment by AminoffZ** [comment]
Looks like a nice way to use Github's API. Seems a bit overkill for this project though 😅. Don't think we need an abstraction layer quite yet but something to keep in mind if the scope of this project changes!

**Reply by mAmineChniti** [reply]
This could be really useful in case this gets even more complicated and (i don't completely understand api.ts) but i think the repo information fetching could be even more simplified, can it be more simplified and if yes and you have some free time can you tell me what exactly it does?
